kong/plugins/config-by-env/config.lua

```diff
@@ -1,100 +1,37 @@
-local inspect = require("inspect")
 local cjson_safe = require "cjson.safe"
 local utils = require "kong.plugins.config-by-env.utils"
 
-local function load_plugin_from_db(key)
-	local row, err = kong.db.plugins:select_by_cache_key(key)
-	if err then
-		return nil, tostring(err)
-	end
-	return row
+local ngx = ngx
+
+local function process_config(conf)
+    local config, err = cjson_safe.decode(conf.config)
+    local env = os.getenv("KONG_ENV")
+    if err then
+        kong.log.err(err)
+        error("Error in parsing config-by-env as table")
+        return nil
+    end
+
+    local final_config = utils.tableMerge(config["default"], config[env])
+    local success, err = utils.traverseTableAndTransformLeaves(final_config, utils.replaceStringEnvVariables)
+    if not success then
+        error(err)
+    end
+
+    kong.log.inspect("Processed Config:")
+    kong.log.inspect(final_config)
+    return final_config
 end
 
-local function load_plugin_from_cache(name)
-	local cache_key = kong.db.plugins:cache_key(name)
-	local opts = {ttl = 0}
-	local plugin, err = kong.core_cache:get(cache_key, opts, load_plugin_from_db, cache_key)
-	if err then
-		kong.log.err(err)
-		return false, {status = 500, message = "Error in loading from cache"}
-	end
-	return plugin
+local function get_config(conf)
+    local config, err = kong.cache:get("config-by-env-final", nil, process_config, conf)
```

Comment:
Why add a ttl to cache? 
This seems like an overhead commuting of parsing config every 30 seconds

Reply:
agreed.